core-web/src/main/java/org/silverpeas/core/webapi/wopi/WopiServerFilter.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2000 - 2020 Silverpeas
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * As a special exception to the terms and conditions of version 3.0 of
+ * the GPL, you may redistribute this Program in connection with Free/Libre
+ * Open Source Software ("FLOSS") applications as described in Silverpeas's
+ * FLOSS exception.  You should have received a copy of the text describing
+ * the FLOSS exception, and it is also available here:
+ * "https://www.silverpeas.org/legal/floss_exception.html"
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.silverpeas.core.webapi.wopi;
+
+import org.silverpeas.core.web.http.HttpRequest;
+import org.silverpeas.core.web.session.UserSessionEvent;
+import org.silverpeas.core.wopi.DefaultWopiUser;
+import org.silverpeas.core.wopi.WopiFileEditionManager;
+import org.silverpeas.core.wopi.WopiLogger;
+import org.silverpeas.core.wopi.WopiUser;
+
+import javax.enterprise.event.Observes;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Enumeration;
+
+import static java.text.MessageFormat.format;
+import static javax.ws.rs.core.Response.Status.fromStatusCode;
+
+/**
+ * @author silveryocha
+ */
+public class WopiServerFilter implements Filter {
```

Comment:
Please, explain what does this filter and for which reason it is required in the case of WOPI